### DISCUSSION TITLE
ImGui_ImplOpenGL3_CreateDeviceObjects failed to compile verted/fragment shader & link shader program

### DISCUSSION THREAD

**Discussion by jmoraispk** [ocornut/imgui#6017] [New Users Build/Link/Run issues ONLY!]
When executing a DearPyGui script, which uses ImGui, I got this error, and it seems related with ImGui. 

""" (...keeps repeating...)
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile vertex shader!
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile fragment shader!
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to link shader program! (with GLSL '#version 130
')
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile vertex shader!
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile fragment shader!
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to link shader program! (with GLSL '#version 130
')
^CTraceback (most recent call last):
  File "/home/joao/Documents/GitHub/DeepSense-viewer/src/main.py", line 597, in <module>
    dpg.render_dearpygui_frame()
  File "/home/joao/miniconda3/envs/joao-stable/lib/python3.9/site-packages/dearpygui/dearpygui.py", line 8877, in render_dearpygui_frame
    return internal_dpg.render_dearpygui_frame(**kwargs)
KeyboardInterrupt
"""

I'm on Ubuntu 22.04 with OpenGL version 4.6. Output of glxinfo | grep "version":
server glx version string: 1.4
client glx version string: 1.4
GLX version: 1.4
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.0.5
OpenGL core profile shading language version string: 4.60
OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.0.5
OpenGL shading language version string: 4.60
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 22.0.5
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
    GL_EXT_shader_implicit_conversions, GL_EXT_shader_integer_mix, 

Any idea of what might be happening?

**Comment by PathogenDavid** [comment]
> ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile vertex shader!

The wording of the shader compilation error message changed back in June 2021 (specifically during 70c603854e8817770c1a4eeb047929eb1b7313a3) so the version of Dear ImGui you're using is quite old.

Additionally this version is prior to [the embedded loader change](https://github.com/ocornut/imgui/issues/4445), which means OpenGL is not automatically loaded on your behalf. Loader failures manifest themselves in different ways, so it's hard to say for sure if it's your problem here.

You should probably just update to a newer version of the package to rule out this problem entirely.

**Reply by jmoraispk** [reply]
Thanks @PathogenDavid. The version of ImGUI I'm using was packaged jointly with DearPyGUI, so I'll talk with them.
